tests/baselines/reference/user/ts-toolbelt.log

```diff
@@ -3,6 +3,12 @@ Standard output:
 ../../../../built/local/lib.dom.d.ts(618,5): error TS2687: All declarations of 'url' must have identical modifiers.
 ../../../../node_modules/@types/node/globals.d.ts(30,5): error TS2687: All declarations of 'url' must have identical modifiers.
 ../../../../node_modules/@types/node/globals.d.ts(30,5): error TS2717: Subsequent property declarations must have the same type.  Property 'url' must be of type 'string | undefined', but here has type 'string'.
+node_modules/ts-toolbelt/out/index.d.ts(3527,92): error TS2321: Excessive stack depth comparing types 'MinusNegative<?, N2, IMap>' and 'MinusNegative<?, N2, IMap>'.
+node_modules/ts-toolbelt/out/index.d.ts(3527,92): error TS2321: Excessive stack depth comparing types 'MinusNegative<N1, ?, IMap>' and 'MinusNegative<N1, ?, IMap>'.
+node_modules/ts-toolbelt/out/index.d.ts(3527,92): error TS2321: Excessive stack depth comparing types 'MinusNegative<N1, N2, ?>' and 'MinusNegative<N1, N2, ?>'.
+node_modules/ts-toolbelt/out/index.d.ts(3527,92): error TS2321: Excessive stack depth comparing types 'MinusPositive<?, N2, IMap>' and 'MinusPositive<?, N2, IMap>'.
+node_modules/ts-toolbelt/out/index.d.ts(3527,92): error TS2321: Excessive stack depth comparing types 'MinusPositive<N1, ?, IMap>' and 'MinusPositive<N1, ?, IMap>'.
+node_modules/ts-toolbelt/out/index.d.ts(3527,92): error TS2321: Excessive stack depth comparing types 'MinusPositive<N1, N2, ?>' and 'MinusPositive<N1, N2, ?>'.
```

Comment:
@millsp you wouldn't happen to know offhand if this can be worked around trivially in `ts-toolbelt` or not, would you? I can look into a workaround for my change to avoid triggering these, but this _probably_ indicates that if you caused these types to get their variance measured before, they'd issue this error.

Reply:
I think (I assume) that it comes from:

https://github.com/millsp/ts-toolbelt/blob/51c9df18bdf0cdeec016ad5715620023f7434897/src/Number/Minus.ts#L20

and

https://github.com/millsp/ts-toolbelt/blob/51c9df18bdf0cdeec016ad5715620023f7434897/src/Number/Minus.ts#L43

which are mistakes. Just for testing, if you can put `number` instead of `N2`, then it should work. Let me know, because there might be other optimizations that I had to work on :) In any case, TypeScript has to move forward, and I will adapt in future releases.

Reply:
Yeah, you can definitely trigger these errors in currently shipping typescript (or at least the current nightly) - just for variance calculation of the type by using it in a comparison of some kind (doesn't even have to make much sense):

```ts
type Box<X> = {f: MinusNegative<Cast<X, Iteration>, Cast<X, Iteration>, Map>};

const x = null as any as Box<number>;
const y: Box<12> = x;
```
`y` will trigger all these `Excessive stack depth` errors. :(

So if you can fix it, you probably should - it might already be causing headaches for your consumers.

Reply:
Will look into it